sympy/matrices/dense.py

```diff
@@ -880,7 +880,10 @@ def row_del(self, i):
         row
         col_del
         """
-        self._mat = self._mat[:i*self.cols] + self._mat[(i + 1)*self.cols:]
+        if i >= -self.rows and i < self.rows:
+            del self._mat[i*self.cols:(i + 1)*self.cols]
+        else:
+            raise IndexError("Index out of range: 'i = %s', valid -%s <= i < %s" % (i, self.rows, self.rows))
```

Comment:
As I found the del operation is much faster compared to the assignment to the self._mat.